akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpChallenge.java

```diff
@@ -15,17 +16,9 @@
     public abstract Map<String, String> getParams();
 
     public static HttpChallenge create(String scheme, String realm) {
-        return new akka.http.scaladsl.model.headers.HttpChallenge(scheme, realm, Util.emptyMap);
+        return akka.http.scaladsl.model.headers.HttpChallenge.apply(scheme, Option.apply(realm));
     }
     public static HttpChallenge create(String scheme, String realm, Map<String, String> params) {
-        return new akka.http.scaladsl.model.headers.HttpChallenge(scheme, realm, Util.convertMapToScala(params));
-    }
-
-    public static HttpChallenge createBasic(String realm) {
-        return create("Basic", realm);
-    }
-
-    public static HttpChallenge createOAuth2(String realm) {
-        return create("Bearer", realm);
+        return akka.http.scaladsl.model.headers.HttpChallenge.apply(scheme, Option.apply(realm), Util.convertMapToScala(params));
```

Comment:
Meh, now I reviewed the bottom half of the PR and see what you mean.
I think I'd keep as is as it'll be easier to fix the signatures to become Option (in scaladsl) when we get the chance to.